ros/src/computing/planning/motion/packages/op_local_planner/nodes/op_behavior_selector/op_behavior_selector_core.cpp

```diff
@@ -0,0 +1,495 @@
+	_nh.getParam("/op_common_params/maxAcceleration", m_CarInfo.max_acceleration);
+	_nh.getParam("/op_common_params/maxDeceleration", m_CarInfo.max_deceleration);
+	m_CarInfo.max_speed_forward = m_PlanningParams.maxSpeed;
+	m_CarInfo.min_speed_forward = m_PlanningParams.minSpeed;
+
+	PlannerHNS::ControllerParams controlParams;
+	controlParams.Steering_Gain = PlannerHNS::PID_CONST(0.07, 0.02, 0.01);
+	controlParams.Velocity_Gain = PlannerHNS::PID_CONST(0.1, 0.005, 0.1);
+	nh.getParam("/op_common_params/steeringDelay", controlParams.SteeringDelay);
+	nh.getParam("/op_common_params/minPursuiteDistance", controlParams.minPursuiteDistance );
+
+	int iSource = 0;
+	_nh.getParam("/op_common_params/mapSource" 			, iSource);
+	if(iSource == 0)
+		m_MapType = PlannerHNS::MAP_AUTOWARE;
+	else if (iSource == 1)
+		m_MapType = PlannerHNS::MAP_FOLDER;
+	else if(iSource == 2)
+		m_MapType = PlannerHNS::MAP_KML_FILE;
+
+	_nh.getParam("/op_common_params/mapFileName" 		, m_MapPath);
+
+	_nh.getParam("/op_behavior_selector/evidence_tust_number", m_PlanningParams.nReliableCount);
+	m_PlanningParams.additionalBrakingDistance = 5.0;
+
+	//std::cout << "nReliableCount: " << m_PlanningParams.nReliableCount << std::endl;
+
+	m_BehaviorGenerator.Init(controlParams, m_PlanningParams, m_CarInfo);
+	m_BehaviorGenerator.m_pCurrentBehaviorState->m_Behavior = PlannerHNS::INITIAL_STATE;
+
+}
+
+void BehaviorGen::callbackGetTwistRaw(const geometry_msgs::TwistStampedConstPtr& msg)
+{
+	m_Twist_raw = *msg;
+}
+
+void BehaviorGen::callbackGetTwistCMD(const geometry_msgs::TwistStampedConstPtr& msg)
+{
+	m_Twist_cmd = *msg;
+}
+
+void BehaviorGen::callbackGetCommandCMD(const autoware_msgs::ControlCommandConstPtr& msg)
+{
+	m_Ctrl_cmd = *msg;
+}
+
+void BehaviorGen::callbackGetCurrentPose(const geometry_msgs::PoseStampedConstPtr& msg)
+{
+	m_CurrentPos = PlannerHNS::WayPoint(msg->pose.position.x, msg->pose.position.y, msg->pose.position.z, tf::getYaw(msg->pose.orientation));
+	bNewCurrentPos = true;
+}
+
+void BehaviorGen::callbackGetVehicleStatus(const geometry_msgs::TwistStampedConstPtr& msg)
+{
+	m_VehicleStatus.speed = msg->twist.linear.x;
+	m_CurrentPos.v = m_VehicleStatus.speed;
+	if(fabs(msg->twist.linear.x) > 0.25)
+		m_VehicleStatus.steer = atan(m_CarInfo.wheel_base * msg->twist.angular.z/msg->twist.linear.x);
+	UtilityHNS::UtilityH::GetTickCount(m_VehicleStatus.tStamp);
+	bVehicleStatus = true;
+}
+
+void BehaviorGen::callbackGetCanInfo(const autoware_msgs::CanInfoConstPtr &msg)
+{
+	m_VehicleStatus.speed = msg->speed/3.6;
+	m_CurrentPos.v = m_VehicleStatus.speed;
+	m_VehicleStatus.steer = msg->angle * m_CarInfo.max_steer_angle / m_CarInfo.max_steer_value;
+	UtilityHNS::UtilityH::GetTickCount(m_VehicleStatus.tStamp);
+	bVehicleStatus = true;
+}
+
+void BehaviorGen::callbackGetRobotOdom(const nav_msgs::OdometryConstPtr& msg)
+{
+	m_VehicleStatus.speed = msg->twist.twist.linear.x;
+	m_CurrentPos.v = m_VehicleStatus.speed ;
+	if(msg->twist.twist.linear.x != 0)
```

Comment:
eq comparison with float/double

Reply:
just to avoid division by zero

Reply:
again see my comment above, you can not compare 2 floating point numbers.